src/uwtools/drivers/jedi.py

```diff
@@ -0,0 +1,134 @@
+        """
+        The driver.
+
+        :param config: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.files_copied(),
+            self.files_linked(),
+            self.configuration_file(),
+            self.runscript(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def validate_only(self):
+        """
+        Validate config.
+        """
+        taskname = self._taskname("validate_only")
+        yield taskname
+        a = asset(None, lambda: False)
+        yield a
+        path = Path(self._driver_config["configuration_file"]["base_file"])
+        executable = Path(self._driver_config["execution"]["executable"])
+        yield [file(executable), file(path)]
+        env = " && ".join(self._driver_config["execution"]["envcmds"])
+        cmd = "{env} && time {x} --validate-only {p} 2>&1".format(env=env, x=executable, p=path)
+        result = run(taskname, cmd)
+        if result.success:
+            logging.info("%s: Config is valid", taskname)
+            a.ready = lambda: True
+
+    @task
+    def configuration_file(self):
+        """
+        The configuration file.
+        """
+        fn = "input.yaml"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        self._create_user_updated_config(
+            config_class=YAMLConfig,
+            config_values=self._driver_config.get("configuration_file", {}),
```

Comment:
Using `.get()` with a `{}` default implies that it's OK for this task to produce an empty YAML file as `input.yaml`. Is that really ok? If the driver needs _some_ file, then this should perhaps be
```
config_values=self._driver_config["configuration_file"],
```
so that the code will at least fail if we somehow make it this far without specifying either `base_file` or `update_values`. (When the schema is fleshed out, that should catch such an error even earlier by making `configuration_file` required; and the `validate_only` task, if made a requirement of other task methods, like `provisioned_run_directory` or `run` as suggested above, might also reject an empty YAML file. But the code here should also reflect the intent that we _must have_ a non-empty YAML file, if that's accurate.)

Reply:
@NaureenBharwaniNOAA, there's an example of this in the MPAS and MPAS Init drivers in their respective `namelist_file()` methods that use a `try`/`except` block. There's also unit tests that check for empty YAML files.